src/modules/projectSearchContext.js

```diff
@@ -0,0 +1,34 @@
+class ProjectSearchContext {
+
+    /**
+     * Create a search context for a specific project. This can be used to search for traces.
+     * 
+     * @param {SessionManager} sessionManager The session manager, used for connections to the Hansken servers
+     * @param {UUID} projectId The project id
+     */
+    constructor(sessionManager, projectId) {
+        this.sessionManager = sessionManager;
+        this.projectId = projectId;
+    }
+
+    /**
+     * Search for traces in a project.
+     * 
+     * @param {string|object} request The query as an HQL query string, or the full REST request for the /search
+     * @returns 
+     */
+    traces = (request = '') => {
+        const searchRequest = typeof request === 'string' ? {query: {human: request}} : request;
+
+        return this.sessionManager.gatekeeper(`/projects/${this.projectId}/traces/search`, {
+            method: 'POST',
+            headers: {
+                'Accept': 'application/json',
+                'Content-Type': 'application/json'
+            },
+            body: JSON.stringify(searchRequest)
```

Comment:
Why do we stringify the body? It feels like this should be handled by the request api

Reply:
I think it is best to stick as much to the Fetch API as possible, instead of adding magic in the sessionManager.fetch

Also, this is the place where we know exactly how to handle the object and how to convert it for the request, right?